library/deleteDrug.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+/**
+ *  @package   OpenEMR
+ *  @link      http://www.open-emr.org
+ *  @author    Sherwin Gaddis <sherwingaddis@gmail.com>
+ *  @copyright Copyright (c )2020. Sherwin Gaddis <sherwingaddis@gmail.com>
+ *  @license   https://github.com/openemr/openemr/blob/master/LICENSE GNU General Public License 3
+ *
+ *
+ */
+
+require_once "../interface/globals.php";
+
+use OpenEMR\Common\Csrf\CsrfUtils;
+
+if (isset($_POST["drugId"])) {
+    if (!CsrfUtils::verifyCsrfToken($_POST["csrf_token_form"])) {
+        CsrfUtils::csrfNotVerified();
+    }
```

Comment:
Personally i'd trim($_POST["drugId"]). Also, our point for csrf is to verify if a valid request so, i'd add an error return back to requester not allowing request to continue.
I've recently noted several errors in core where this or decrypt fails and continues making hard to track down. I'll be opening an issue on this Brady.

I'd also either wrap sql fetch in a try block or check return somehow so at least user knows of success or not. I'm bad about this too:)

Reply:
The user will know of the success immediately because the screen will reload. If the user selects cancel nothing happens. But upon no errors and successfully delete, the screen reloads showing the entry has been removed.

Reply:
Do note that csrfNotVerified() does an exit, so no return or exception to process:
https://github.com/openemr/openemr/blob/master/src/Common/Csrf/CsrfUtils.php#L71-L80
(if that error is showing up in standard use when it should not, then there is a bug that needs to be addressed)

Reply:
> Do note that csrfNotVerified() does an exit,

Couldn't let this pass. Still, in the context of an ajax request, a dying php function will not propagate that back to the requesting call.
This, while stopping the requested transaction in php, many times won't be propagated back causing the calling JS to most likely just happily continue on. Unless the calling ajax/fetch is trapping for a returned success indication from request.

Now if csrfNotVerified() threw an exception, that would get back to requester as a status 500 causing a fallthrough to ajax fail and not success.

Reply:
There should be only 3 ways for csrf to fail:
1. bad actor attempting csrf attack
2. lazy bad/bored actor trying to run script directly via browser path
3. session leak (ie. forget to use top.restoreSession() with multiple openemr instances being used in same browser)

Just don't think we need to be "nice" about this to the user since it should never happen unless somebody is doing things they are not supposed to be doing, they are being hacked, or there is a bug in the code. This does make me recall our session craziness, which we will likely need to address at some point since I am willing to bet that browsers in the not so distant future will force cookie_httponly on, which will cause OpenEMR to come crashing down (the portal will stay standing though since it supports cookie_httponly nicely :) ).
https://github.com/openemr/openemr/issues/2556

Reply:
Mainly pointing this out as general workflow issue and something we should look out for as we all get lazy about this.
We have many of these scenarios throughout core. They can cause unseen errors that may propagate elsewhere.
Just saying. Not arguing the point:)